controllers/storagecluster/cephcluster.go

```diff
@@ -461,9 +461,12 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string, nodeCount int, s
 				rookCephv1.KeyOSD: systemNodeCritical,
 			},
 			Resources: newCephDaemonResources(sc),
-			ContinueUpgradeAfterChecksEvenIfNotHealthy: true,
+			ContinueUpgradeAfterChecksEvenIfNotHealthy: !resourceProfileChangeisHappening,
 			LogCollector: logCollector,
 			Labels: rookCephv1.LabelsSpec{
+				rookCephv1.KeyMgr:        rookCephv1.Labels{defaults.ODFResourceProfileKey: sc.Spec.ResourceProfile},
+				rookCephv1.KeyMon:        rookCephv1.Labels{defaults.ODFResourceProfileKey: sc.Spec.ResourceProfile},
+				rookCephv1.KeyOSD:        rookCephv1.Labels{defaults.ODFResourceProfileKey: sc.Spec.ResourceProfile},
```

Comment:
@iamniting from the UI perspective, when creating a new cluster the customer will always have to choose a profile and the backend will receive it.
For existing clusters that do not have a `resourceProfile` set yet, we should determine if they were deployed as `minimal` i.e. setting these `ResourceRequirements` (that no longer apply once this feature is merged):
https://github.com/red-hat-storage/odf-console/blob/master/packages/odf/types/common.ts#L42-L74

Reply:
@alfonsomthd We should not determine if they are in minimal mode or not as we are setting the values for 5 daemons. There are fewer chances that we will have the same values as the `minimal`. Moreover, if someone has set some values that are not default they should always be treated as custom instead of assigning any profile to them automatically.

If someone has not set any values we should treat it as balanced which is default now.

Reply:
Hey @alfonsomthd , The minimal mode that was being deployed by the UI was nothing but specifying the daemon resources manually in the storagecluster CR. With this profile mechanism also we respect the manual values added in the CR first ahead of the profiles. So if there are any clusters present that had deployed on that minimal mode from UI their clusters will work fine even after the upgrade as their daemons will still use the values that are present in the CR.

Reply:
@iamniting What I want to point out is that if we deployed as minimal was due to the cluster having very low resources. If now we blindly apply Balanced to that cluster, chances are that we could face pod scheduling issues, right?

Reply:
We will not make it balanced. It will still stay as it is. As custom resources were set there.